src/util/transactionUtils.ts

```diff
@@ -71,3 +77,211 @@ export function getTransactionDisplayPayload({
+  const sourceTransactionFinalized = stepEvaluator(transaction.block, bestBlockFinalized);
+  // 2. When the target chain knows about a bigger source block number we infer that transaction block was realayed to target chain.
+  const blockFinalityRelayed = stepEvaluator(transaction.block, bestBridgedFinalizedBlockOnTarget);
+  // 3.1 We read the latest received nonce of the target chain rpc state call.
+  // 3.2 With the value obtained we compare it with the transaction nonce, if the value is bigger or equal then means target chain is aware about this nonce.
+  const messageDelivered = stepEvaluator(transaction.messageNonce, nonceOfBestTargetBlock);
+  // 4.1 *
+  // 4.2 We match the transaction nonce with the current nonce of the best finalized target block
+  const messageFinalizedOnTarget = stepEvaluator(transaction.messageNonce, nonceOfFinalTargetBlock);
+
+  // 5. Once the source chain is confirms through the latestReceivedNonceOnSource, that target chain is aware about the message nonce, the transaction is completed.
+  const sourceConfirmationReceived = stepEvaluator(transaction.messageNonce, latestReceivedNonceOnSource);
+  const onChainCompleted = (value: boolean) => completionStatus(value) === TransactionStatusEnum.COMPLETED;
+
+  // 4.1 * We catch the best block on target related to the message delivery.
+  let updateDeliveryBlock = false;
+  if (messageDelivered && !deliveryBlock) {
+    updateDeliveryBlock = true;
+  }
+
+  let setTransactionComplete = false;
+  if (sourceConfirmationReceived) {
+    setTransactionComplete = true;
+    dispatchMessage(
+      MessageActionsCreators.triggerSuccessMessage({
+        message: `Transaction: ${shortenItem(transaction.blockHash)} is completed`
+      })
+    );
+  }
+
+  const updatedSteps = [
+    {
+      id: 'test-step-include-message-block',
+      chainType: sourceChain,
+      label: 'Include message in block',
+      labelOnChain: transaction.block,
+      status: completionStatus(!!transaction.block)
+    },
+    {
+      id: 'test-step-finalized-block',
+      chainType: sourceChain,
+      label: 'Finalize block',
+      status: completionStatus(sourceTransactionFinalized)
+    },
+    {
+      id: 'test-step-relay-block',
+      chainType: targetChain,
+      label: 'Relay block',
+      status: completionStatus(blockFinalityRelayed)
+    },
+    {
+      id: 'test-step-deliver-message-block',
+      chainType: targetChain,
+      label: 'Deliver message in target block',
+      labelOnChain: onChainCompleted(messageDelivered) && deliveryBlock,
+      status: completionStatus(messageDelivered)
+    },
+    {
+      id: 'test-step-finalized-message',
+      chainType: targetChain,
+      label: 'Finalize message',
+      status: completionStatus(messageFinalizedOnTarget)
+    },
+    {
+      id: 'test-step-confirm-delivery',
+      chainType: sourceChain,
+      label: 'Confirm delivery',
+      status: completionStatus(sourceConfirmationReceived)
+    }
+  ];
```

Comment:
```suggestion
  const updatedSteps = [
    step(1, sourceChain, completionStatus(!!transaction.block), transaction.block),
    step(2, sourceChain, completionStatus(sourceTransactionFinalized)),
    step(3, targetChain, completionStatus(blockFinalityRelayed)),
    step(4, targetChain, completionStatus(messageDelivered), onChainCompleted(messageDelivered) && deliveryBlock),
    step(5, targetChain, completionStatus(messageFinalizedOnTarget)),
    step(6, sourceChain, completionStatus(sourceConfirmationReceived))
  ];
```

Reply:
i was thinking to do something like this , but as the PR scope got challenging i decided to leave it as it is and address it in next PR.
Since you took the time to solve this i will definitely accept this. Thanks!